zebra/kernel_netlink.c

```diff
@@ -1103,11 +1103,12 @@ void kernel_init(struct zebra_ns *zns)
 		zlog_err("Can't set %s socket error: %s(%d)",
 			 zns->netlink_cmd.name, safe_strerror(errno), errno);
 
+	assert(zns->netlink.sock >= 0);
```

Comment:
let's just remove the assert? The above calls into netlink_socket fail out if socket creation fails and we exit.  So this is impossible